games-simulation/openrct2/openrct2-0.2.4.ebuild

```diff
@@ -0,0 +1,182 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake-utils xdg-utils
+
+MY_PN="OpenRCT2"
+MY_PN_OBJ="objects"
+MY_PN_RPL="replays"
+MY_PN_TS="title-sequences"
+MY_PV_OBJ="1.0.12"
+MY_PV_RPL="0.0.4"
+MY_PV_TS="0.1.2c"
+
+DESCRIPTION="An open source re-implementation of Chris Sawyer's RollerCoaster Tycoon 2"
+HOMEPAGE="https://openrct2.org/"
+SRC_URI="
+	https://github.com/${MY_PN}/${MY_PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	https://github.com/${MY_PN}/${MY_PN_OBJ}/releases/download/v${MY_PV_OBJ}/${MY_PN_OBJ}.zip -> ${PN}-${MY_PN_OBJ}-${MY_PV_OBJ}.zip
+	https://github.com/${MY_PN}/${MY_PN_TS}/releases/download/v${MY_PV_TS}/${MY_PN_TS}.zip -> ${PN}-${MY_PN_TS}-${MY_PV_TS}.zip
+	test? ( https://github.com/${MY_PN}/${MY_PN_RPL}/releases/download/v${MY_PV_RPL}/${MY_PN_RPL}.zip -> ${PN}-${MY_PN_RPL}-${MY_PV_RPL}.zip )
+"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="dedicated libressl +lightfx +multiplayer +opengl test +truetype twitch"
+REQUIRED_USE="dedicated? ( !lightfx multiplayer !opengl !truetype !twitch )"
+
+DEPEND_RDEPEND="
+	dev-libs/icu:=
+	dev-libs/jansson:=
```

Comment:
Why do you need this subslot specification? jansson doesn't specify a subslot and no other ebuild in the tree depends on it with a subslot:

```
$ qgrep 'dev-libs/jansson' | wc -l
88
```

vs

```
$ qgrep 'dev-libs/jansson:=' | wc -l
0
```

Reply:
Nope, you are entirely correct. Dropped.

Reply:
Allright, as requested in IRC, let me try to clarify. It was a mistake to add subslot specification in first place. I must have overlooked here. Normally, I first look at a dependecy `foo`, it it specifies (sub) slots. Since in this case, `dev-libs/jansson` doesn't even have a slot at all, it does not make sense to specify it here. My change in first place said to trigger a rebuild on slot change, but as there is no slot (ok, in reality, it's SLOT=0), it won't ever happen. So as a consequence, I dropped it.

Reply:
I added the subslot specification in first place, because it's linked. If I look after emerging openrct2, I can see. that `/usr/lib64/libopenrct2.so` is linked against `libjansson.so.4`. That means for me, if such SONAME, for example due version, would change (and potentially could also break ABI), it would break the `openrct2` ebuild, once `dev-libs/jansson`, installs a new version (for example `libjansson.so.5` with changed SONAME. Without subslot specification, a user would need to run `emerge @preserved-rebuild`, in order to trigger a rebuild, if such sitation would happen. Otherweise, portage will handle that automatically. I am aware, that current version of `dev-libs/jansson` does not have any slots (aka default SLOT="0"), but who knows, that could change in future. Currently, it wouldn't harm to have it added, but also not specifying wouldn't harm, as for now.

Reply:
> I added the subslot specification in first place, because it's linked. If I look after emerging openrct2, I can see. that `/usr/lib64/libopenrct2.so` is linked against `libjansson.so.4`. That means for me, if such SONAME, for example due version, would change (and potentially could also break ABI), it would break the `openrct2` ebuild, once `dev-libs/jansson`, installs a new version (for example `libjansson.so.5` with changed SONAME. Without, a user would need to run `emerge @preserved-rebuild`, in order to trigger a rebuild, if such sitation would happen. Otherweise, portage will handle that automatically. I am aware, that current version of `dev-libs/jansson` does not have any slots (aka default `SLOT="0"`), but who knows, that could change in future. Currently, it wouldn't harm to have it added in case for future, but also not specifying wouldn't harm, as for now.

You can't know what the future use of subslots will be for a given package. You can't assume that the subslots will reflect the ABI SONAME. That is usually the case but there is no rule like that.

Reply:
> You can't know what the future use of subslots will be for a given package. You can't assume that the subslots will reflect the ABI SONAME. That is usually the case but there is no rule like that.

Thats right, I can't now for 100% for sure. So what's the best practice? For packages, which currently are not slotted, don't set any slot operator?

> My understanding is that subslot resolving is what makes portage slow. So you shouldn't deliberately break it this way. Besides, why go against common practices like this? You have added multiple subslot binders on dependencies that don't have subslots.

>https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Subslots
>kinda says you shouldn't use them when not available. @gktrk maybe QA team needs to be consulted here?

Same questions goes here. I changed the operators now, so only got included where subslots do exists. There is only a question for `libpng`. This had once a few weeks ago multiple slots, but all slots expect 0 have been migrated to `libpng-compat`. Should I still keep the slot here in case? It drop it too?

Reply:
```
23:47 <+gokturk> can I have the QA team's opinion on using subslot dependency spec ':=' on packages that don't define a subslot? the argument for it is that if there ever is a new subslot, the dependent package will automatically rebuilt. my counter-argument is that it's not always guaranteed that a future subslot will reflect the ABI soname, and ':=' shouldn't be used unless the package clearly defined subslots.
23:47 <+gokturk> ^
23:47 <+gokturk> !proj qa
23:47 <+willikins> gokturk: (qa@gentoo.org) amynka, dilfridge, mgorny, pinkbyte, soap, ulm, zlogene
01:20 <@ulm> gokturk: we shouldn't clutter our dependency specs with it
01:21 <@ulm> i.e. only use slot dependencies if the package is slotted
01:23 <@ulm> and we cannot know in advance for what purpose a slot would be introduced, so the := may well turn out to be wrong, might well be that it should be :0 or :0= or :* instead
01:23 <@ulm> or even another slot
```

Reply:
Thank you for clarifying. So based on that, I only left `dev-libs/icu`, `dev-libs/libzip`, `media-libs/libpng` and `dev-libs/libressl` / `dev-libs/openssl` with an operator, as those packages do have sub slots. Since `sys-libs/zlib` is according to juipong wrongly with an subslot and should not be used, i dropped that.